apps/passport-client/src/dispatch.ts

```diff
@@ -942,3 +948,36 @@ async function promptToAgreePrivacyNotice(state: AppState, update: ZuUpdate) {
     });
   }
 }
+
+/**
+ * Formats the user's data as JSON, then creates a data URL link and
+ * simulates a click on that link in order to trigger a download.
+ */
+async function exportData(state: AppState, _update: ZuUpdate) {
+  const { self: user, pcds, subscriptions } = state;
+
+  // Since we already use this data for remote sync, we know that it's
+  // sufficient for loading an account on to a new device.
+  const { serializedStorage, storageHash } = await serializeStorage(
+    user,
+    pcds,
+    subscriptions
+  );
+
+  // Data in a data URL must be Base64-encoded
+  const data = Buffer.from(JSON.stringify(serializedStorage)).toString(
+    "base64"
+  );
```

Comment:
question: is this robust to thousands of frogs? is there a way for it to be robust to thousands of frogs? what are the limits of this approach? It seems likely that it is technically possible for a webpage to create and download gigabytes of data as a file. Could we do it in a way that is robust to that situation, so we don't have to revisit this in the future?

Reply:
Firefox has the smallest limit here, which is a maximum of 32MB, which is comfortably larger than the local storage size limit of 10MB.

Reply:
My production PCD collection with 193 frogs is 488KB. On that basis, I would need maybe ~13,000 frogs before the PCD collection reaches 32MB in size - but at this point I would already have exceeded the local storage limit of 5MB-10MB.

We could switch to using IndexedDB in the future, which has larger storage limits, and in this case we might end up storing too much data locally for the data URL solution to work. Saving larger files would mean either a) using the File System API or b) using a service worker endpoint as the download URL.